src/Storage/Storage.Management/Blob/DisableAzureStorageBlobDeleteRetentionPolicy.cs

```diff
@@ -0,0 +1,122 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.Storage
+{
+    using Microsoft.Azure.Commands.Management.Storage.Models;
+    using Microsoft.Azure.Management.Storage;
+    using Microsoft.Azure.Management.Storage.Models;
+    using System;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+    /// <summary>
+    /// Modify Azure Storage service properties
+    /// </summary>
+    [Cmdlet("Disable", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + StorageBlobDeleteRetentionPolicy, SupportsShouldProcess = true, DefaultParameterSetName = AccountNameParameterSet), OutputType(typeof(PSDeleteRetentionPolicy))]
+    public class DisableAzStorageBlobDeleteRetentionPolicyCommand : StorageBlobBaseCmdlet
+    {
+
+        /// <summary>
+        /// AccountName Parameter Set
+        /// </summary>
+        private const string AccountNameParameterSet = "AccountName";
+
+        /// <summary>
+        /// Account object parameter set 
+        /// </summary>
+        private const string AccountObjectParameterSet = "AccountObject";
+
+        /// <summary>
+        /// BlobServiceProperties ResourceId  parameter set 
+        /// </summary>
+        private const string PropertiesResourceIdParameterSet = "BlobServicePropertiesResourceId";
+
+        [Parameter(
+          Position = 0,
+          Mandatory = true,
+          HelpMessage = "Resource Group Name.",
+         ParameterSetName = AccountNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = "Storage Account Name.",
+           ParameterSetName = AccountNameParameterSet)]
+        [ResourceNameCompleter("Microsoft.Storage/storageAccounts", nameof(ResourceGroupName))]
+        [Alias(AccountNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountName { get; set; }
+
+        [Parameter(Mandatory = true,
+            HelpMessage = "Storage account object",
+            ValueFromPipeline = true,
+            ParameterSetName = AccountObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSStorageAccount StorageAccount { get; set; }
```

Comment:
If you have this parameter set (that is a parameter set that takes PSStorageAccount), you should not have ValueFromPipelineByProeprtyName for the other properties.

Reply:
As we discussed, the only parameter has ValueFromPipelineByProeprtyName os "-ResourceId" and this is acceptable. 
As we discussed, I have change the parameter name in all 4 new cmdlets to "ResourceId", and all will accept both Storage account resourceId and Blob Service ResourceId.